app/models/katello/concerns/pulp_database_unit.rb

```diff
@@ -20,8 +20,14 @@ def content_type
         self::CONTENT_TYPE
       end
 
+      def pulp3_support?
+        type = Katello::RepositoryTypeManager.find_repository_type content_type
+        type.pulp3_plugin && SmartProxy.pulp_master!.capabilities('Pulp3').try(:include?, type.pulp3_plugin)
```

Comment:
I think you might be mixing up two things here, the repository type which the pulp3_plugin indicates and the content type.  For file these are the same (file repos and file content), but for other types they aren't.